handlers/search.go

```diff
@@ -417,51 +417,34 @@ func FindValidatorIndicesByEth1Address(search string) (types.SearchValidatorsByE
 	if len(eth1AddressHash) == 20 {
 		// if it is an eth1-address just search for exact match
 		err = db.ReaderDb.Select(result, `
-			SELECT from_address, COUNT(*), ARRAY_AGG(validatorindex) validatorindices FROM (
-				SELECT 
-					DISTINCT ON(validatorindex) validatorindex,
-					ENCODE(from_address::bytea, 'hex') as from_address,
-					DENSE_RANK() OVER (PARTITION BY from_address ORDER BY validatorindex) AS validatorrow,
-					DENSE_RANK() OVER (ORDER BY from_address) AS addressrow
-				FROM eth1_deposits
-				INNER JOIN validators ON validators.pubkey = eth1_deposits.publickey
-				WHERE from_address = $1
-			) a 
-			WHERE validatorrow <= $2 AND addressrow <= 10
-			GROUP BY from_address
-			ORDER BY count DESC`, eth1AddressHash, searchValidatorsResultLimit)
-	} else if len(eth1AddressHash) <= 16 {
-		// if the lenght is less then 32 use byte-wise comparison
-		err = db.ReaderDb.Select(result, `
-			SELECT from_address, COUNT(*), ARRAY_AGG(validatorindex) validatorindices FROM (
+			SELECT from_address_text, COUNT(*), ARRAY_AGG(validatorindex) validatorindices FROM (
 				SELECT 
 					DISTINCT ON(validatorindex) validatorindex,
-					ENCODE(from_address::bytea, 'hex') as from_address,
-					DENSE_RANK() OVER (PARTITION BY from_address ORDER BY validatorindex) AS validatorrow,
-					DENSE_RANK() OVER (ORDER BY from_address) AS addressrow
+					from_address_text,
+					DENSE_RANK() OVER (PARTITION BY from_address_text ORDER BY validatorindex) AS validatorrow,
+					DENSE_RANK() OVER (ORDER BY from_address_text) AS addressrow
 				FROM eth1_deposits
 				INNER JOIN validators ON validators.pubkey = eth1_deposits.publickey
-				WHERE from_address LIKE $1 || '%'::bytea
+				WHERE from_address_text = $1
```

Comment:
You removed the %, don't get why?
But you added it in line 443.
Is this cause by the rework?

Reply:
No its the git highlighting that is misleading.
the `from_address_text = $1` is from the first if statement where we do an exact match search if the eth1AddressHash has the exact length. 

I remove the second if statement as it is no longer needed with the switch to the new from_address_text column.

And in the else there is the wildcard search.